test/unit/bson_regex.test.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const { expect } = require('chai');
+const { BSONRegExp } = require('../../src/index');
+
+describe('BSONRegExp', () => {
+  describe('bsonRegExp option', () => {
+    it('should respond with BSONRegExp class with option passed to db', async function () {
+      let client;
+      try {
+        // create and connect to client
+        client = this.configuration.newClient();
+        await client.connect();
+
+        const db = client.db('a', { bsonRegExp: true });
+        const collection = db.collection('b');
+
+        await collection.insertOne({ regex: new BSONRegExp('abc', 'imx') });
+        const res = await collection.findOne({ regex: new BSONRegExp('abc', 'imx') });
+
+        expect(res).has.property('regex').that.is.instanceOf(BSONRegExp);
+      } finally {
+        await client.close();
+      }
+    });
+
+    it('should respond with BSONRegExp class with option passed to collection', async function () {
+      let client;
+      try {
+        // create and connect to client
+        client = this.configuration.newClient(); // bsonRegex
+        await client.connect();
+
+        const db = client.db('a');
+        const collection = db.collection('b', { bsonRegExp: true });
```

Comment:
The setup is nearly the same for each test, can we share some of this logic in a before / beforeEach.
Or I think we can make a loop that tests each of these, with only one implementation:

```javascript
const options = { bsonRegexp: true };
for (const passOptionsTo of ['client', 'db',  'collection', 'operation']) {
  it(`should respond with BSONRegExp class with option passed to ${passOptionsTo}`, async function () {
        client = this.configuration.newClient(passOptionsTo === 'client' ? options : undefined);
        await client.connect();

        const db = client.db('a', passOptionsTo === 'db' ? options : undefined);
        const collection = db.collection('b', passOptionsTo === 'collection' ? options : undefined);

        await collection.insertOne({ regex: new BSONRegExp('abc', 'imx') });
        const res = await collection.findOne(
          { regex: new BSONRegExp('abc', 'imx') },
          passOptionsTo === 'operation' ? options : undefined
        );
        
        // ....
  });
}
```
How does this look? needs some clean up but this is the idea.

Reply:
Looks good to me.